### PR TITLE
Add more extensive documentation about ws_connbuf

### PR DESCRIPTION
**Pull request by neithernut**
This PR adds more extensive documentation for the connection interface.

### CODE REVIEW

**Comment by neithernut** [comment]
Documentation for `ws_connector` is still missing, but you can review the `ws_connbuf` docu.

**Comment by matthiasbeyer** [comment]
ACK so far. I compiled the documentation and the formatting looks right.

**Comment by neithernut** [comment]
`ws_connector` should be ready now, too.

File: src/connection/connector.h [hunk 1]
```diff
@@ -33,9 +33,30 @@
...
+ * Both of them are meant to be accessed from external modules and can also be
+ * thought of as an "inbox" and an "outbox".
+ *
+ * If an entity wished to read data from an connection, it will call
+ * `ws_connector_read()`, which performs a `read()` on the embedded file
+ * descriptor, filling up `inbuf`.
+ * After invoking that function, the reading entity may read data from the
+ * `inbuf` and `ws_connbuf_discard()` that data.
+ *
+ * If an entity wishes to write to an connection, it may feed that data to the
```
Comment (xVin) [inline src/connection/connector.h]: a connection

File: src/connection/connector.h [hunk 2]
```diff
@@ -33,9 +33,30 @@
...
+ * Connector
+ *
+ * A `ws_connector` wraps a file descriptor which may be used to send or receive
+ * data.
+ * For this purpose, it features two embedded `ws_connbuf`s, of which one is an
+ * input buffer and one is an output buffer.
+ * Both of them are meant to be accessed from external modules and can also be
+ * thought of as an "inbox" and an "outbox".
+ *
+ * If an entity wished to read data from an connection, it will call
```
Comment (xVin) [inline src/connection/connector.h]: wishes?
a connection

File: src/connection/connbuf.h [hunk 3]
```diff
@@ -38,7 +38,54 @@
...
+ * When `read()`ing and `write()`ing to a connection, a programmer has to take
+ * care of the buffering of incomplete content.
+ *
+ * E.g. a series of bytes read from a file descriptor may or may not contain
+ * a message which may be deserialized by a parser in one run.
+ * If a streaming parser is used, residual content, which may be parsed some
+ * time in the future, may be left in the buffer.
+ * Similarly, a `write()` may not be able to send every last byte of the buffer.
+ *
+ * The intent behind a connection buffer is to ease reading and writing from and
```
Comment (xVin) [inline src/connection/connbuf.h]: ease --> facilitate

File: src/connection/connbuf.h [hunk 4]
```diff
@@ -38,7 +38,54 @@
...
+ * to connections by taking care of the leftovers of invocations of either the
+ * dezerializer or `write()`.
+ * This is _not_ implemented as a ring-buffer, since we're assuming relatively
+ * small payloads, which may be processed once at a time.
+ * Instead, the idea is to have one buffer, which logically is divided into
```
Comment (xVin) [inline src/connection/connbuf.h]: , which is logically divided

File: src/connection/connbuf.h [hunk 5]
```diff
@@ -38,7 +38,54 @@
...
+ * three areas:
+ *  * data which may be written to a destination or deserialized
+ *  * reserved space, to which a `read()` or a serialized may write to
```
Comment (xVin) [inline src/connection/connbuf.h]: it seems like a noun is missing after "serialized"

Comment (neithernut) [inline src/connection/connbuf.h]: Nope, it's a typo in `serializer`.

File: src/connection/connbuf.h [hunk 6]
```diff
@@ -38,7 +38,54 @@
...
+ *
+ * E.g. a series of bytes read from a file descriptor may or may not contain
+ * a message which may be deserialized by a parser in one run.
+ * If a streaming parser is used, residual content, which may be parsed some
+ * time in the future, may be left in the buffer.
+ * Similarly, a `write()` may not be able to send every last byte of the buffer.
+ *
+ * The intent behind a connection buffer is to facilitate reading and writing
+ * from and to connections by taking care of the leftovers of invocations of
+ * either the dezerializer or `write()`.
```
Comment (TheNeikos) [inline src/connection/connbuf.h]: de _s_ erializer

**Comment by TheNeikos** [comment]
Other than that it's looking good!

**Comment by neithernut** [comment]
Any more comments?

**Pull request merged**